### PR TITLE
fix(mongodb-compass): Do not use postinstall to rebuild dependencies

### PR DESCRIPTION
**Pull request by gribnoysup**
`packages/compass` postinstall script currently rebuilds application dependencies with electron-rebuild which causes them to be not compatible with node version that some test suites are using, this causes flakiness when running tests on GitHub CI (see #2277 and some other PRs where tests were failing with the similar error)

This is safe to remove and will not affect Compass application packaging because when `hadron-build` packages the application it will rebuild native modules using `electron-rebuild` after reinstalling application dependencies (see [packages/hadron-build/commands/release.js#L366-L394](https://github.com/mongodb-js/compass/blob/208d064756484c6fec128e8b9764738e338ef1d0/packages/hadron-build/commands/release.js#L366-L394))

For some additional context the reason it's only causing issues some of the time is that in some test suites we rebuild those native modules to work in electron and then back to work in node after tests are finished, so if the stars are aligned correctly and the first test suite rebuilds native modules back to the node runtime, everything works fine.

Test runs in this PR will not really show that it fixes the issue as you need this specific order of tests running to make the issue appear, but I tried it locally against #2277 and all tests were passing with the fix applied on top of the branch

### CODE REVIEW

**Comment by gribnoysup** [comment]
Build step failures are not caused by anything changed in this patch and related to the issues outlined in #2279

File: .github/workflows/check-test.yaml [hunk 1]
```diff
@@ -57,10 +57,11 @@ jobs:
...
-        # linting and dependency checking is relevant only for 
-        run: npm run check -- --stream --since $MAIN_BRANCH_NAME --exclude-dependents
+        # linting and dependency checking is relevant only for packages where
+        # the source code was changed
+        run: npm run check-evergreen -- --stream --since $MAIN_BRANCH_NAME --exclude-dependents
         shell: bash
 
       - name: Run Tests
-        run: npm run test -- --stream --since $MAIN_BRANCH_NAME
+        run: npm run test-evergreen -- --stream --since $MAIN_BRANCH_NAME
```
Comment (mcasimir) [inline .github/workflows/check-test.yaml:66]: Do we want to also rename these tasks to `test-ci` / `check-ci` at this point?

**Review by mcasimir — APPROVED** [review]
Nice!

File: .github/workflows/check-test.yaml (hunk 1 above)
Comment (gribnoysup) [reply .github/workflows/check-test.yaml:66]: Sure, will do

**Pull request merged**